java/tests_unitaires/fr/TESTS_UNITAIRES_JAVA.md

```diff
@@ -0,0 +1,171 @@
+# Les tests unitaires en Java
+
+Grâce aux différentes librairies, nous allons pouvoir rendre nos tests simples à comprendre et à écrire.
+
+## Mise en Contexte
+
+Voici la classe ``Calculator`` que nous allons tester :
+
+```java
+public class Calculator {
+
+    private final int a;
+    private final NumberProvider numberProvider;
+
+    public Calculator(int a, NumberProvider numberProvider) {
+        this.a = a;
+        this.numberProvider = numberProvider;
+    }
+
+    public int add(int b) {
+        return a + b;
+    }
+
+    public float divide(int b) {
+
+        if (b == 0) {
+            throw new IllegalArgumentException("Denominator must be non null.");
+        }
+
+        return (float) a / b;
+    }
+
+    public int addToNumber() {
+        return a + numberProvider.provideNumber();
+    }
+
+}
+
+public interface NumberProvider {
+
+    int provideNumber();
+
+}
+```
+
+C'est une classe qui contient des méthodes très simples à tester, le but étant seulement de découvrir les tests en java.
+
+NumberProvider est une interface qui fourni simplement un nombre, on pourrait imaginer une implémentation par exemple fournissant un nombre aléatoire.
+L'utilité fondamentale de cette interface n'est pas forcément incroyable mais elle nous permettra d'utiliser des mocks.
+
+Les points à retenir de cette classe Calculator pour les tests sont :
+
+ - Le field ``numberProvider`` qui nous permettra de d'expérimenter les mocks.
+ - Le test de la méthode divide qui lance une exception si le dénominateur est nul.
+
+## Ecriture des tests
+
+### Préparation de la classe de test
+
+Avant tout, on commence par créer un field ``calculator`` qui sera utilisé pour les tests. Il sera recréé à chaque test grâce à l'annotation ``@Before`` de la méthode ``setUp`` :
+
+```java
+import org.junit.Before;
+import org.mockito.Mockito;
+
+public class CalculatorTest {
+
+    private Calculator calculator;
+    private NumberProvider numberProvider;
+
+    @Before
+    public void setUp() {
+        this.numberProvider = Mockito.mock(NumberProvider.class);
+        this.calculator = new Calculator(10, numberProvider);
+    }
+
+}
+```
+
+La méthode ``Mockito.mock(T)`` retourne une instance de ``T`` que l'on pourra manipuler pour maîtriser les retours des méthodes de ce mock et donc nos tests.
```

Comment:
I believe the correct signature is `Mockito.mock(Class<T>)`, according to your usage